src/migrate_backups.py

```diff
@@ -0,0 +1,249 @@
+"""
+Migrate our backups from a complete copy of CrunchyBridge's rolling 10 day window
+to a snapshot of backups of the first and third Saturday's of each month.
+
+The backup directories already contain the reduced backups. This needs to
+convert it over to the new location.
+
+Iterate through folders in our backup folders
+    cluster/
+    ├─ archive/ # This is the WAL that is stored while the backup is created
+    │   └─ backup_stanza/
+    │      ├─ 15-1/
+    │      │   └─ 0000000100000821/
+    │      │      └─ 00000001000008210000001D.lz4
+    │      └─ archive.info
+    └─ backup/ # This is the physical backup created each day
+       └─ backup_stanza/
+          ├─ backup.history/
+          ├─ backup.info
+          ├─ backup.info.copy
+          └─ 20230101-010000F/
+             ├─ backup.manifest
+             ├─ backup.manifest.copy
+             └─ pg_data/
+
+Look through the backup.manifest file for
+backup-archive-start and backup-archive-stop
+Find all files between these two values.
+
+"""
+import argparse
+import os
+from datetime import datetime
+from typing import Optional
+
+import sentry_sdk
+from dateutil.relativedelta import relativedelta
+from dotenv import load_dotenv
+
+from src.delete_backups import CRUNCHYBRIDGE_BACKUP_PATTERN
+from src.s3 import get_s3
+
+# ENV Variables
+load_dotenv()
+
+ASPIRE_AWS_ACCESS_KEY_ID = os.getenv("ASPIRE_AWS_ACCESS_KEY_ID")
+ASPIRE_AWS_SECRET_ACCESS_KEY = os.getenv("ASPIRE_AWS_SECRET_ACCESS_KEY")
+SENTRY_DSN = os.getenv("SENTRY_DSN")
+
+
+def lsn_in_range(start_hex, end_hex):
+    for value_decimal in range(int(start_hex, 16), int(end_hex, 16) + 1):
+        value_hex = hex(value_decimal)[2:].upper()
+        yield value_hex.zfill(24)
+
+
+def parse_manifest(body) -> (str, str):
+    """
+    Parse start and stop LSN from backup.manifest
+
+    The typical format of these files is:
+
+
+        [backrest]
+        backrest-format=5
+        backrest-version="2.45"
+
+        [backup]
+        backup-archive-start="00000001000008210000001D"
+        backup-archive-stop="00000001000008210000001F"
+        backup-label="20230916-010001F"
+        backup-lsn-start="821/1D000028"
+        backup-lsn-stop="821/1F0B73F0"
+    """
+    start = None
+    stop = None
+    for line in body.read().decode("utf-8").splitlines():
+        if line.startswith("backup-archive-start"):
```

Comment:
nitpick: I think we could add `not start and` to this line so it doesn't have to do the `startswith` check if `start` is already set:

`if not start and line.startswith("backup-archive-start"):`

Same thing for the `stop` check below.